File: app/src/test/java/com/cvv/fanstaticapps/randomticker/activities/MainActivityTest.java
package com.cvv.fanstaticapps.randomticker.activities;

import android.content.Intent;
import android.widget.SeekBar;
import android.widget.TextView;

import com.cvv.fanstaticapps.randomticker.R;
import com.cvv.fanstaticapps.randomticker.helper.TimeSeekBarChangeListener;

import org.junit.Test;
import org.robolectric.shadows.ShadowApplication;

import java.util.Locale;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNull;

/**
 * Created by Carla
 * Date: 05/02/2018
 * Project: RandomTicker
 */
public class MainActivityTest extends RobolectricBaseTest<MainActivity> {

    public MainActivityTest() {
        super(MainActivity.class);
    }

    @Test
    public void testTimer_NotCreatedWithMinBiggerThanMax() throws Exception {
        setSeekBarValues(11, 4, 0, 0);

        clickStartButton();

        Intent actual = ShadowApplication.getInstance().getNextStartedActivity();
        assertNull(actual);
    }

    @Test
    public void testTimer_CreatedWithMinSmallerThanMax() throws Exception {
        setSeekBarValues(2, 3, 10, 59);

        clickStartButton();

        Intent expectedIntent = new Intent(activity, KlaxonActivity.class);
        Intent actual = ShadowApplication.getInstance().getNextStartedActivity();
        assertEquals(expectedIntent.getComponent(), actual.getComponent());

    }

    private void clickStartButton() {
        getView(R.id.start).performClick();
    }

    private void setSeekBarValues(int minMinutes, int minSeconds, int maxMinutes, int maxSeconds) {
        SeekBar minMinSeekBar = getView(R.id.min_min);
        SeekBar minSecSeekBar = getView(R.id.min_sec);
        SeekBar maxMinSeekBar = getView(R.id.max_min);
        SeekBar maxSecSeekBar = getView(R.id.max_sec);
        maxMinSeekBar.setProgress(maxMinutes);
        maxSecSeekBar.setProgress(maxSeconds);
        minMinSeekBar.setProgress(minMinutes);
        minSecSeekBar.setProgress(minSeconds);

        TextView minDisplayText = getView(R.id.min_value);
        TextView maxDisplayText = getView(R.id.max_value);
        String expectedMinValue = formatExpectedValue(minMinutes, minSeconds);
        String expectedMaxValue = formatExpectedValue(maxMinutes, maxSeconds);

        assertEquals(expectedMinValue, minDisplayText.getText().toString());
        assertEquals(expectedMaxValue, maxDisplayText.getText().toString());
    }

    private String formatExpectedValue(int minutes, int seconds) {
        return String.format(Locale.getDefault(), TimeSeekBarChangeListener.TIME_FORMAT, minutes, seconds);
    }

}

File: app/src/test/java/com/cvv/fanstaticapps/randomticker/activities/MainActivityTest.kt
package com.cvv.fanstaticapps.randomticker.activities

import android.content.Intent
import android.view.View
import android.widget.TextView
import com.cvv.fanstaticapps.randomticker.R
import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertNull
import org.junit.Test
import org.robolectric.shadows.ShadowApplication

/**
 * Created by carvaq
 * Date: 05/02/2018
 * Project: RandomTicker
 */
class MainActivityTest : RobolectricBaseTest<MainActivity>(MainActivity::class.java) {

    @Test
    fun `max values are smaller than min values`() {
        setValues(11, 4, 0, 0)

        clickStartButton()

        val actual = ShadowApplication.getInstance().nextStartedActivity
        assertNull(actual)
    }

    @Test
    fun `max values are bigger than min values and don't overflow`() {
        setValues(2, 3, 480, 450)

        val maxMinTextView = getView<TextView>(R.id.maxMin)
        val maxSecTextView = getView<TextView>(R.id.maxSec)

        assertEquals("240", maxMinTextView.text.toString())
        assertEquals("59", maxSecTextView.text.toString())

        clickStartButton()

        val expectedIntent = Intent(activity, KlaxonActivity::class.java)
        val actual = ShadowApplication.getInstance().nextStartedActivity
        assertEquals(expectedIntent.component, actual.component)

    }

    private fun clickStartButton() {
        getView<View>(R.id.start).performClick()
    }

    private fun setValues(minMinutes: Int, minSeconds: Int, maxMinutes: Int, maxSeconds: Int) {
        val minMinTextView = getView<TextView>(R.id.minMin)
        val minSecTextView = getView<TextView>(R.id.minSec)
        val maxMinTextView = getView<TextView>(R.id.maxMin)
        val maxSecTextView = getView<TextView>(R.id.maxSec)
        minMinTextView.text = minMinutes.toString()
        minSecTextView.text = minSeconds.toString()
        maxMinTextView.text = maxMinutes.toString()
        maxSecTextView.text = maxSeconds.toString()

    }

}